src/pybind/mgr/diskprediction/agent/predict/prediction.py

```diff
@@ -0,0 +1,299 @@
+from __future__ import absolute_import
+
+import datetime
+import json
+import os
+import time
+
+from .. import BaseAgent
+from ...common.db import DB_API
+
+
+PREDICTION_FILE = '/var/tmp/disk_prediction.json'

```

Comment:
What is the purpose of this file?  

The mgr usually can't store data in files in a useful way since it can fail over to another daemon on another host at any time.

Reply:
This file is temp file to store the prediction result, like a cache file. It can quickly response the prediction result when user use diskprediction command to get prediction result. If mgr fail over to another node, the diskprediction plugin will create the temp file when it get prediction file from the server.

Reply:
While `/var/tmp` probably is a safe bet, I'm not a big fan of hard-coded paths.
Would it make sense to use `tempfile.gettempdir()` here?

Or is there a point in storing this information in the MONs database instead? I wonder if @jcsp has an opinion on that?

Reply:
If we are recreating this data on a mgr failover, then it should also be possible to recreate it on a restart: the active mgr tends to be a different node after restart the active one anyway.

I'm assuming that the disk prediction state would be available in memory, so you wouldn't be reading this file in before each user command? (Currently this patch doesn't ever call the read function)

If this data was coming from an external source then storing a local copy in the mon might make sense, but I think this plugin is storing it in a RADOS pool, so storing it both places doesn't help (presumably if the data is big enough to need to be in rados, then it is too big for the mon).

Reply:
@jcsp Thanks. Good idea, I will change my code to store prediction data in the memory.